r/R/dplyr.R

```diff
@@ -182,7 +182,7 @@ dim.arrow_dplyr_query <- function(x) {
     # Query on in-memory Table, so evaluate the filter
     # Don't need any columns
     x <- select.arrow_dplyr_query(x, NULL)
-    rows <- nrow(compute.arrow_dplyr_query(x))
+    rows <- nrow(as_arrow_table(x))
```

Comment:
This is because manipulating metadata for a table with no rows will cause the size to be updated to 0 x 0.


``` r
mtcars |> arrow::arrow_table() |> dplyr::select(NULL) |> arrow::as_arrow_table()
#> Table
#> 32 rows x 0 columns
#>
#>
#> See $metadata for additional Schema metadata
mtcars |> arrow::arrow_table() |> dplyr::select(NULL) |> arrow::as_arrow_table() |> dplyr::ungroup()
#> Table
#> 0 rows x 0 columns
#>
#>
#> See $metadata for additional Schema metadata
```

<sup>Created on 2022-10-07 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

I don't know if this (handling of tables with no rows) is the problem.